google_shopping_functions.js

```diff
@@ -7,8 +7,38 @@ var getItemsCount = function(itemData){
 
 //
 //DEFINE FUNCTIONS HERE
-//
+var getItems = function (objectData){
+	return objectData.items
+};
+var getItemsByBrand = function(items, brandName){
+	var itemArray = [];
+	for (var i = 0; i < items.length; i++){
+		if (items[i].product.brand.toLowerCase() === brandName.toLowerCase()){
+			itemArray.push(items[i]);
+		}
+	}
+	return itemArray;
+};
 
+var getItemsByAuthor = function(items, authorName){
+	var itemArray = [];
+	for (var i = 0; i < items.length; i++){
+		if (items[i].product.author.name.toLowerCase().indexOf(authorName.toLowerCase()) !== -1) {
+			itemArray.push(items[i]);
+		}
+	}
+	return itemArray;
+};
+
+var getAvailableProducts = function(items){
```

Comment:
Here, you could also pass your availability string as a parameter, to make the function check for different availability types if you wanted.

```var getProductsByAvailability = function(items, status) {```
```...```
```if (items[i].product.inventories[0].availability === status) {```